python/ray/rllib/models/catalog.py

```diff
@@ -121,7 +136,11 @@ def get_preprocessor(cls, env, options=dict()):
             env.observation_space.shape = ()
 
         env_name = env.spec.id
-        obs_shape = env.observation_space.shape
+
+        if isinstance(env.observation_space, list):
+            obs_shape = env.observation_space[0].shape
+        else:
+            obs_shape = env.observation_space.shape
```

Comment:
What if we did np.stack in the RayEnv / preprocessor so that the shape becomes a tensor like
`[len(original_space), original_space[0]]`?

Reply:
Hm maybe the right thing to do is to support tuple spaces in RLlib in general. We need this for Carla to work anyways.

Reply:
Nevermind, the above probably won't work if the spaces are irregularly shaped.